webapp/channels/src/components/admin_console/content_flagging/user_multiselector/user_multiselector.tsx

```diff
@@ -0,0 +1,124 @@
+
+import {debounce} from 'mattermost-redux/actions/helpers';
+import {getMissingProfilesByIds, searchProfiles} from 'mattermost-redux/actions/users';
+import {getUsersByIDs} from 'mattermost-redux/selectors/entities/users';
+
+import type {GlobalState} from 'types/store';
+
+import {UserProfilePill} from './user_profile_pill';
+
+import {UserOptionComponent} from '../../content_flagging/user_multiselector/user_profile_option';
+import {LoadingIndicator} from '../../system_users/system_users_filters_popover/system_users_filter_team';
+
+import './user_multiselect.scss';
+
+export type AutocompleteOptionType<T> = {
+    label: string | ReactElement;
+    value: string;
+    raw?: T;
+}
+
+type Props = {
+    id: string;
+    className?: string;
+    onChange: (selectedUserIds: string[]) => void;
+    initialValue?: string[];
+    hasError?: boolean;
+}
+
+export function UserMultiSelector({id, className, onChange, initialValue, hasError}: Props) {
+    const dispatch = useDispatch();
+    const {formatMessage} = useIntl();
+    const initialDataLoaded = useRef<boolean>(false);
+
+    useEffect(() => {
+        const fetchInitialData = async () => {
+            await dispatch(getMissingProfilesByIds(initialValue || []));
+            initialDataLoaded.current = true;
+        };
+
+        if (initialValue && !initialDataLoaded.current) {
+            fetchInitialData();
+        }
+    }, [dispatch, initialValue]);
+
+    const initialUsers = useSelector((state: GlobalState) => getUsersByIDs(state, initialValue || []));
+    const selectInitialValue = initialUsers.
+        filter((userProfile) => Boolean(userProfile)).
+        map((userProfile: UserProfile) => ({
+            value: userProfile.id,
+            label: userProfile.username,
+            raw: userProfile,
+        } as AutocompleteOptionType<UserProfile>));
+
+    const userLoadingMessage = useCallback(() => formatMessage({id: 'admin.userMultiSelector.loading', defaultMessage: 'Loading users'}), [formatMessage]);
+    const noUsersMessage = useCallback(() => formatMessage({id: 'admin.userMultiSelector.noUsers', defaultMessage: 'No users found'}), [formatMessage]);
+    const placeholder = formatMessage({id: 'admin.userMultiSelector.placeholder', defaultMessage: 'Start typing to search for users...'});
+
+    const searchUsers = useMemo(() => debounce(async (searchTerm: string, callback) => {
+        try {
+            const response = await dispatch(searchProfiles(searchTerm, {page: 0}));
+            if (response && response.data && response.data.length > 0) {
+                const users = response.data.
+                    filter((userProfile) => !userProfile.is_bot).
+                    map((user) => ({
+                        value: user.id,
+                        label: user.username,
+                        raw: user,
+                    }));
+
+                callback(users);
+            }
+
+            callback([]);
```

Comment:
shouldn't this callback call be in an else statement?

Reply:
Done.